k8s/k8s.go

```diff
@@ -1137,50 +1160,144 @@ func (k *k8sOps) isPVCShared(pvc *v1.PersistentVolumeClaim) bool {
 
 // PVCs APIs - END
 
+// Snapshot APIs - BEGIN
+
+func (k *k8sOps) CreateSnapshot(snap *snap_v1.VolumeSnapshot) (*snap_v1.VolumeSnapshot, error) {
+	var result snap_v1.VolumeSnapshot
+	if err := k.snapClient.Post().
+		Name(snap.Metadata.Name).
+		Resource(snap_v1.VolumeSnapshotResourcePlural).
+		Namespace(snap.Metadata.Namespace).
+		Body(snap).
+		Do().Into(&result); err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+func (k *k8sOps) DeleteSnapshot(snap *snap_v1.VolumeSnapshot) error {
+	return k.snapClient.Delete().
+		Name(snap.Metadata.Name).
+		Resource(snap_v1.VolumeSnapshotResourcePlural).
+		Namespace(snap.Metadata.Namespace).
+		Do().Error()
+}
+
+func (k *k8sOps) ValidateSnapshot(snap *snap_v1.VolumeSnapshot) error {
+	t := func() (interface{}, error) {
+		var result snap_v1.VolumeSnapshot
+		if err := k.snapClient.Get().
+			Name(snap.Metadata.Name).
+			Resource(snap_v1.VolumeSnapshotResourcePlural).
+			Namespace(snap.Metadata.Namespace).
+			Do().Into(&result); err != nil {
+			return "", err
+		}
+
+		for _, condition := range result.Status.Conditions {
+			if condition.Type == snap_v1.VolumeSnapshotConditionReady && condition.Status == v1.ConditionTrue {
```

Comment:
Instead of trying it for 5 minutes shouldn't we just stop checking if we get VolumeSnapshotConditionError as true

Reply:
Update the retry task to stop if the task function returns false for retry